### ISSUE TITLE
Parser: Better comments

### ISSUE THREAD

**Issue by Mitch-Siegel** [Mitch-Siegel/customSystem#10]
Now that I'm switching around test code much more often, I'm realizing it would be really nice to have block comments. Given this, it also makes sense to transition to c-style `//` comments. 
Note - block comments will require detection in both `trimWhitespace()` and `scan()` given that they can appear arbitrarily.

- [x]  Change single-line comment token from `#` to `//`
- [x] Implement block comment parsing in `trimWhitespace()`
- [ ] Implement block comment check in `scan()`

**Comment by Mitch-Siegel** [comment]
Block comments implemented in commit b5f08945f12bba60bb5e9d5958ad6f2c5c44069d. Checking for comments in `scan()` is unnecessary because `trimwhitespace()` will always be called before starting the scan, and block comments can't go in the middle of tokens.
